rust/lancedb/resources/remote-catalog-open-api.yaml

```diff
@@ -0,0 +1,284 @@
+      parameters:
+        - $ref: '#/components/parameters/dbName'
+        - $ref: '#/components/parameters/startAfter'
+        - $ref: '#/components/parameters/limit'
+      responses:
+        200:
+          description: Table list response
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/TableList'
+
+    post:
+      tags: [Table]
+      summary: Create new table
+      parameters:
+        - $ref: '#/components/parameters/dbName'
+      requestBody:
+        required: true
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/CreateTableRequest'
+      responses:
+        201:
+          description: Table created
+          headers:
+            Location:
+              schema: {type: string}
+
+    delete:
+      tags: [ Table ]
+      summary: Delete all tables
+      parameters:
+        - $ref: '#/components/parameters/dbName'
+      responses:
+        204:
+          description: All tables deleted
+
+
+  /v1/databases/{dbName}/tables/{tableName}:
+    get:
+      tags: [Table]
+      summary: Get table metadata
+      parameters:
+        - $ref: '#/components/parameters/dbName'
+        - $ref: '#/components/parameters/tableName'
+      responses:
+        200:
+          description: Table metadata
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/TableMetadata'
+
+    put:
+      tags: [Table]
+      summary: Update table data
+      parameters:
+        - $ref: '#/components/parameters/dbName'
+        - $ref: '#/components/parameters/tableName'
+      requestBody:
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/TableUpdate'
+      responses:
+        200:
+          description: Update successful
+
+    delete:
+      tags: [ Table ]
+      summary: Delete a specific table
+      parameters:
+        - $ref: '#/components/parameters/dbName'
+        - $ref: '#/components/parameters/tableName'
+      responses:
+        204:
+          description: Table deleted
+
+  /v1/databases/{dbName}/tables/{tableName}/name:
+    put:
+      tags: [ Table ]
+      summary: Rename table
+      parameters:
+        - $ref: '#/components/parameters/dbName'
+        - $ref: '#/components/parameters/tableName'
+      requestBody:
+        content:
+          application/json:
+            schema:
+              type: object
+              required: [ newName ]
+              properties:
+                newName: { type: string }
+      responses:
+        204:
+          description: Table renamed
+
+
+components:
+  parameters:
+    dbName:
+      name: dbName
+      in: path
+      required: true
+      schema: {type: string}
+
+    tableName:
+      name: tableName
+      in: path
+      required: true
+      schema: {type: string}
+
+    startAfter:
+      name: startAfter
```

Comment:
I think we should describe more about the behavior of this parameter. If we use this parameter, this means the response returned by the server must be in increasing lexicographical order. If the server cannot support such feature, it should fail the operation with unsupported error code.